Assignment4/angelux101-a180068/task07.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1-yF16MXfbs8l6dAZ5kUbCUZNtrqsYwGE
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+# TO DO
+ns = Namespace("http://somewhere#")
+
+from rdflib.plugins.sparql import prepareQuery
+q1 = prepareQuery('''
+    SELECT ?sub WHERE { 
+    ?sub rdfs:subClassOf ns:Person. 
+    }
+    ''',
+  initNs = {"ns": Namespace("http://somewhere#"), "rdfs": RDFS}
+)
+
+# Visualize the results
+for r in g.query(q1):
+  print(r.sub)
+
+for s,p,o in g.triples((None,RDFS.subClassOf, ns.Person)):
+  print(s)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+# TO DO
+
+for s,p,o in g.triples((None,RDF.type, ns.Person)):
+  print(s)
+
+
+q2 = prepareQuery('''
+    SELECT ?p WHERE { 
+    ?p rdf:type ns:Person. 
+    }
+    ''',
+  initNs = {"ns": Namespace("http://somewhere#"), "rdfs": RDFS}
+)
```

Comment:
This query is only printing the instances that are of type Person, not the entities that are of type person or of type any subclass of person